source/Audio.cpp

```diff
@@ -255,6 +292,111 @@ void Audio::Update(const Point &listenerPosition)
 
 
 
+void Audio::UpdateMusic(PlayerInfo &player, Track::GameState state)
+{
+	if(!isInitialized)
+		return;
+
+	// Music defined by planet or system.
+	if(state == Track::GameState::LANDED)
+	{
+		if(!player.GetPlanet()->MusicName().empty())
+		{
+			PlayMusic(player.GetPlanet()->MusicName());
+			return;
+		}
+	}
+	else if(player.GetSystem() && !player.GetSystem()->MusicName().empty())
+	{
+		PlayMusic(player.GetSystem()->MusicName());
+		return;
+	}
+
+	// Wait for the time set by the track.
+	if(currentTrack->IsFinished() && currentPlaylistTrack)
+	{
+		if(!isWaiting && currentPlaylistTrack->Wait() > 0 && !finishedWaiting)
+		{
+			isWaiting = true;
+			currentWait = currentPlaylistTrack->Wait();
+			onEnd = chrono::high_resolution_clock::now();
+		}
+		if(isWaiting)
+		{
+			chrono::duration<double> elapsed = chrono::high_resolution_clock::now() - onEnd;
+			if(elapsed.count() >= currentWait)
+			{
+				finishedWaiting = true;
+				isWaiting = false;
+			}
+			else
+				return;
+		}
+	}
+
+	// If the current playlists conditions are not matching anymore, search a new one.
+	bool currentPlaylistValid = currentPlaylist ?
+		currentPlaylist->MatchingConditions(player) : false;
+	// The track has to be updated if the current track is finished or the playlist is not matching
+	// anymore.
+	if(currentTrack->IsFinished() || !currentPlaylistValid)
```

Comment:
If I'm reading this code correctly, this could just be split up into two consecutive if blocks?

Reply:
done